feat: toggle transparent background

JosXa opened the pull request:
## Summary

- Introduce a tri-state transparency policy (`auto | on | off`) for the TUI theme, persisted via `theme_transparent` and resolved inside `resolveTheme(theme, mode, transparency)`.
- Add the [`theme.transparency`](packages/tui/src/app.tsx) command to the command palette. Its title reports the current policy and, in `auto`, the resolved effective state (e.g. `Transparency: Auto (on)`). Running it cycles `auto -> on -> off -> auto`.
- Expose `theme.transparent` on the resolved theme as the *effective* runtime state (`background.a < 1`), so any consumer can ask whether the active background is transparent without having to interpret the policy.
- Register [`theme_transparency`](packages/tui/src/config/keybind.ts) in the typed keybind config and map it to `theme.transparency`, matching the neighboring theme bindings.

Closes [#11866](https://github.com/anomalyco/opencode/issues/11866).

## Why

Background transparency is useful when the terminal already provides its own transparency (e.g. Ghostty `background-opacity`, kitty, alacritty), and individual themes increasingly want to express transparency intent themselves. The merged [PR #8467](https://github.com/anomalyco/opencode/pull/8467) already makes the generated `system` theme transparent by default for exactly this reason.

A simple boolean toggle conflicts with that direction: when the active theme is `system`, the resolved background is already transparent, so a label like *Enable transparency* is misleading, and toggling *off* later cannot undo a theme-defined transparent background. The right primitive is a small policy:

- `auto` respects the theme's own background alpha, including `system` being transparent by default.
- `on` forces the resolved background to alpha `0` while preserving the original RGB.
- `off` forces any non-opaque resolved background to alpha `1` while preserving the original RGB, so transparent themes can be made opaque again.

The command palette communicates this directly: in `auto` it shows `Transparency: Auto (on)` or `Transparency: Auto (off)` depending on the active theme, and in the forced states it shows `Force on` / `Force off`.

## Compatibility

- Backwards compatible with the previous boolean: persisted `theme_transparent=true` is read as `"on"`; `false` or any unknown value is read as `"auto"` (the new default).
- Plays well with the existing system-theme transparency from [PR #8467](https://github.com/anomalyco/opencode/pull/8467): `auto` no longer overrides what the theme defines.

## Tests

- Added `resolveTheme transparency policy respects auto, on, and off` in [`packages/tui/test/theme.test.ts`](packages/tui/test/theme.test.ts), covering:
  - Default themes stay opaque under `auto` and become alpha-`0` (with original RGB preserved) under `on`.
  - The generated `system` theme is transparent under `auto`, and becomes opaque (with original RGB preserved) under `off`.
  - Partially transparent backgrounds (`a = 0.5`) are reported as transparent under `auto` and forced to alpha `1` under `off`.

Verified locally:

- `bun test packages/tui/test/theme.test.ts` passes.
- `bun typecheck` in `packages/tui` passes.

## Screenshots

N/A – the change is a TUI-only behavior/state change. The visible difference is the command palette title for `theme.transparency`, e.g.:

- `Transparency: Auto (on)` when the active theme defines a transparent background.
- `Transparency: Auto (off)` when the active theme defines an opaque background.
- `Transparency: Force on` and `Transparency: Force off` when the policy is forced.

JosXa commented:
https://github.com/user-attachments/assets/475d5804-59d0-4b1b-9d42-8e96ccb4a2ad

JosXa commented on packages/opencode/src/cli/cmd/tui/context/theme.tsx:
```diff
@@ -213,10 +214,17 @@ function resolveTheme(theme: ThemeJson, mode: "dark" | "light") {
+  if (transparent) {
+    resolved.background = RGBA.fromInts(0, 0, 0, 0)
+    // NOTE: Could alternatively apply an alpha channel to the theme's base background color
+    // instead of forcing full transparency, allowing for adjustable opacity levels
```
Not sure how to model this with the TUI though.

I saw https://github.com/anomalyco/opentui/blob/main/packages/core/src/renderables/Slider.ts, but it doesn't appear to exposed to Solid yet

ariane-emory commented:
Dynamically switching the text for this menu item from "Enable transparency" to "Disable transparency" to indicate its current state would be a nice touch.

Commit:
feat(tui): add transparent background toggle

Adds a new command to toggle transparent background for terminal emulators
that support transparency. The toggle text dynamically updates based on
the current transparency state.

Closes #11866

Commit:
fix(tui): align run theme transparency typing

Commit:
feat(tui): add transparency policy
